public/app/plugins/datasource/loki/querybuilder/components/LokiQueryBuilderContainer.tsx

```diff
@@ -26,11 +26,14 @@ export interface State {
 export function LokiQueryBuilderContainer(props: Props) {
   const { query, onChange, onRunQuery, datasource } = props;
   const [state, dispatch] = useReducer(stateSlice.reducer, {
-    expr: '',
-    visQuery: {
-      labels: [],
-      operations: [{ id: '__line_contains', params: [''] }],
-    },
+    expr: query.expr,
+    // Use initial visual query only if empty query
+    visQuery: query.expr
+      ? undefined
+      : {
+          labels: [],
+          operations: [{ id: '__line_contains', params: [''] }],
+        },
```

Comment:
not sure I understand this, query.expr is always defined 

https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/querybuilder/state.ts#L45

Reply:
I shouldd probably change the comment from empty query to query with empty string. 

When we are deciding about initial state,  if `query.expr === ''`, it means that user is going to start to write new query and we want to show `line_contains` operation. If we have `query.expr !== ''` and we switch to visual query builder, we want to simply create visual query from query.expr and we don't any need initial state for visual query.

I like this option, because if we would go different road and create default `query.expr` for visual query builder, we would have it as `{} |= ""` (so we have `line_contains` operation). And even if we didn't filled the `line_contains` `|= ""` would always be there. And when you switch between visual editor -> code editor-> visual editor, you would get error  because `{} |= ""` is not valid query. 

This way, it is there and available for user who is starting to write new query, but it doesn't produce unnecessary query parts for the user who don't want to use it.

Reply:
Does it make sense?

Reply:
ah I understand, yes query.expr === '' ?  makes sense then

Reply:
Yes! I can change it to make it more clear.